probe-rs-tools/src/bin/probe-rs/cmd/run/mod.rs

```diff
@@ -514,3 +523,99 @@ fn poll_rtt<S: Write + ?Sized>(
 
     Ok(out.had_data)
 }
+
+struct SemihostingPrinter {
+    stdout_open: bool,
+    stderr_open: bool,
+}
+
+impl SemihostingPrinter {
+    const STDOUT: u32 = 1;
+    const STDERR: u32 = 2;
+
+    pub fn new() -> Self {
+        Self {
+            stdout_open: false,
+            stderr_open: false,
+        }
+    }
+
+    pub fn handle(
+        &mut self,
+        command: SemihostingCommand,
+        core: &mut Core<'_>,
+    ) -> Result<(), Error> {
+        match command {
+            SemihostingCommand::Open(request) => {
+                let path = request.path(core)?;
+                if path == ":tt" {
+                    match request.mode().as_bytes()[0] {
+                        b'w' => {
```

Comment:
This is actually part of the `SH_EXT_STDOUT_STDERR` - but we don't yet support extensions. I imagine this PR is far from the semihosting implementation that we should have, but we'll have to start somewhere.